crowbar_framework/app/helpers/installer/upgrades_helper.rb

```diff
@@ -58,9 +58,7 @@ def upgrade_error_flash(message)
     end
 
     def upgrade_ha_repo_needed?
-      return true if Proposal.where(barclamp: "pacemaker").first
-
-      false
+      !!Proposal.where(barclamp: "pacemaker").first
```

Comment:
I agree with @rsalevsky. You need to play with type coercion. A more clear option is implement `.exists?` in the `Proposal` object, or whatever returns the `where`